llvm/utils/TableGen/SearchableTableEmitter.cpp

```diff
@@ -646,11 +671,23 @@ void SearchableTableEmitter::collectTableEntries(
     if (auto RecordTy = dyn_cast<RecordRecTy>(Field.RecType)) {
       if (IntrinsicClass && RecordTy->isSubClassOf(IntrinsicClass))
         Field.IsIntrinsic = true;
-      else if (InstructionClass && RecordTy->isSubClassOf(InstructionClass))
+      else if (InstructionClass && RecordTy->isSubClassOf(InstructionClass)) {
         Field.IsInstruction = true;
+        SawInstructionField = true;
+      }
     }
   }
 
+  // Build instruction-to-int map to check for contiguous instruction values.
+  // These are the same values emitted by InstrInfoEmitter. Do this on demand
+  // only after it is known that there are definitely instruction fields.
+  if (SawInstructionField && InstrEnumValueMap.empty()) {
+    CodeGenTarget Target(Records);
+    unsigned Num = 0;
+    for (const CodeGenInstruction *Inst : Target.getInstructionsByEnumValue())
```

Comment:
I think we should add a `CodeGenTarget::getInstrToIntMap` (or other names?) to get this instr->int mapping, so that we can reuse these logics and sync with other emittters (like you said, `InstrInfoEmitter::emitEnums`).

Reply:
As you suggested, I have moved this map into `CodeGenTarget::InstrToIntMap` and added a lookup method `CodeGenTarget::getInstrIntValue`. The map is now populated in `CodeGenTarget::computeInstrsByEnum` right after the sort that creates the order in the first place. 

Overall this is cleaner and simplified the patch as well. Thanks again for the suggestions.

In a follow-up patch, as you alluded to, I could use the new query in `InstrInfoEmitter::emitEnums` (and elsewhere if applicable) to reuse/sync the logic.